### ISSUE TITLE
MQTT client fails to connect on MacOS

### ISSUE THREAD

**Issue by alvarolobato** [HASwitchPlate/openHASP#598]
### Perform all steps below and tick them with [x]
- [x ] Check the related part of the [Documentation](https://www.openhasp.com/)
- [ x] Update openHASP to the latest version
- [ x] Reproduce the issue and describe all steps

### Describe the bug
Connection to MQTT fails on master with "client error", it works well in 0.6.3
```
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 363
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 369
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 375
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 379
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 294
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 300
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 315
[10:42:01.763][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 333
[10:42:01.967][252504/252504  0] MQTT: Failed to connect, return code -1

```
I tested with old versions of Paho and it doesn't seem related to changes in the library, something seems to have been changed in how the connection is performed in MacOS from OpenHasp.

### To Reproduce
- Build from master on MacOS
- Either configure you MQTT server in the config file or use command line parameters
- Connection will always fail

### Expected behavior
I expect it to connect to the MQTT server
### Screenshots or video
Logs with Paho tracing enabled
```

Command-line arguments:
  argv[0]   ./program
  argv[1]   --mqtthost
  argv[2]   localhost
  argv[1]   --mqtthost
{"mqtt":{"host":"localhost"}}
[10:47:39.285][    0/    0  0] MQTT: host => localhost

        open____ _____ _____ _____
          |  |  |  _  |   __|  _  |
          |     |     |__   |   __|
          |__|__|__|__|_____|__|
        Home Automation Switch Plate
        Open Hardware edition v0.7.0-rc9

[10:47:39.285][    0/    0  0] MAIN: resolution 240 x 320
[10:47:39.285][    0/    0  0] MAIN: pre setup
[10:47:39.285][258680/259296  1] MAIN: Sysname:  Darwin
[10:47:39.285][258680/259296  1] MAIN: Nodename: Alvaros-MBP.local
[10:47:39.285][258680/259296  1] MAIN: Release:  23.1.0
[10:47:39.285][258680/259296  1] MAIN: Version:  Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
[10:47:39.285][258680/259296  1] MAIN: Machine:  arm64
[10:47:39.285][258680/259296  1] MAIN: Processor  : unknown
[10:47:39.285][258680/259296  1] MAIN: CPU freq.  : 0 MHz
[10:47:39.285][258680/259296  1] TFT : Starting...
[10:47:39.395][258680/259240  1] TFT : Driver     : SDL2
[10:47:39.395][258680/259240  1] TFT : SDL Version: v2.28.3
[10:47:39.395][258680/259240  1] TFT : DMA        : Disabled
[10:47:39.395][258680/259240  1] TFT : Started
[10:47:39.395][258680/259240  1] MAIN: Sysname:  Darwin
[10:47:39.395][258680/259240  1] MAIN: Nodename: Alvaros-MBP.local
[10:47:39.395][258680/259240  1] MAIN: Release:  23.1.0
[10:47:39.395][258680/259240  1] MAIN: Version:  Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
[10:47:39.395][258680/259240  1] MAIN: Machine:  arm64
[10:47:39.395][258680/259240  1] MAIN: Processor  : unknown
[10:47:39.395][258680/259240  1] MAIN: CPU freq.  : 0 MHz
[10:47:39.395][258680/259240  1] LVGL: Starting...
[10:47:39.395][258680/259240  1] LVGL: Version    : 7.11.0
[10:47:39.395][258680/259240  1] LVGL: lv_init: already inited
[10:47:39.395][258680/259240  1] LVGL: MEM size   : 262144
[10:47:39.395][258680/259240  1] LVGL: VFB size   : 131072
[10:47:39.395][258680/259112  1] LVGL: Filesystem : Enabled
[10:47:39.395][258680/258952  1] LVGL: LV_FS_PC ready
[10:47:39.395][258680/258944  1] LVGL: /littlefs/config.json
[10:47:39.395][258680/258952  1] HASP: TEST Opening config.json from FS failed 12
[10:47:39.395][258680/258952  1] FONT: FreeType Disabled
[10:47:39.395][257632/257688  1] GUI : Initialize Cursor
[10:47:39.395][257072/257080  1] LVGL: Started
[10:47:39.395][257072/257080  1] MAIN: src/main_sdl2.cpp 126
[10:47:39.395][257072/257080  1] MSGR: Starting...
[10:47:39.395][257072/257080  1] MSGR: Started
[10:47:39.395][253920/254736  1] HASP: Custom theme loaded
[10:47:39.395][253920/254576  1] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253920/254576  1] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253920/254416  1] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253920/254416  1] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253920/254248  1] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253920/254248  1] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253920/254080  1] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253920/254080  1] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253920/253920  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253920/253920  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253752/253752  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253752/253752  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253584/253584  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253584/253584  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253416/253416  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253416/253416  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253248/253248  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253248/253248  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][253080/253080  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][253080/253080  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][252912/252912  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][252912/252912  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][252744/252744  0] HASP: src/hasp/hasp_page.cpp - 165
[10:47:39.395][252744/252744  0] HASP: src/hasp/hasp_page.cpp - 172
[10:47:39.395][252744/252744  0] HASP: Loading .\pages.jsonl from disk...
[10:47:39.395][252744/252744  0] HASP: Loaded .\pages.jsonl from disk
[10:47:39.395][252744/252744  0] HASP: Changing page to 1
[10:47:39.395][252504/252504  0] MAIN: src/main_sdl2.cpp 131
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 363
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 369
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 375
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 379
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 294
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 300
=========================================================
                   Trace Output
Product name: Eclipse Paho Synchronous MQTT C Client Library
Version: @CLIENT_VERSION@
Build level: @BUILD_TIMESTAMP@
=========================================================
19700101 010000.000 (7987781312)  (1)> Socket_outInitialize:141
19700101 010000.000 (7987781312)   (2)> SocketBuffer_initialize:106
19700101 010000.000 (7987781312)   (2)< SocketBuffer_initialize:114 (0)
19700101 010000.000 (7987781312)  (1)< Socket_outInitialize:166
19700101 010000.000 (7987781312)  (1)> MQTTStrncpy:1085
19700101 010000.000 (7987781312)  (1)< MQTTStrncpy:1095
19700101 010000.000 (7987781312)  (1)> MQTTStrncpy:1085
19700101 010000.000 (7987781312)  (1)< MQTTStrncpy:1095
19700101 010000.000 (7987781312)  (1)> Thread_create_sem:227
19700101 010000.000 (7987781312)  (1)< Thread_create_sem:245 (-1)
19700101 010000.000 (7987781312)  (1)> Thread_create_sem:227
19700101 010000.000 (7987781312)  (1)< Thread_create_sem:245 (-1)
19700101 010000.000 (7987781312)  (1)> Thread_create_sem:227
19700101 010000.000 (7987781312)  (1)< Thread_create_sem:245 (-1)
19700101 010000.000 (7987781312)  (1)> Thread_create_sem:227
19700101 010000.000 (7987781312)  (1)< Thread_create_sem:245 (-1)
19700101 010000.000 (7987781312)  (1)> MQTTPersistence_create:56
19700101 010000.000 (7987781312)  (1)< MQTTPersistence_create:104 (0)
19700101 010000.000 (7987781312)  (1)> MQTTPersistence_initialize:119
20231222 104739.395 (7987781312)  (1)< MQTTPersistence_initialize:127 (0)
20231222 104739.395 (7987781312)  (1)> MQTTPersistence_restoreMessageQueue:877
20231222 104739.395 0 queued messages restored for client Alvaros-MBP.local
20231222 104739.395 (7987781312)  (1)< MQTTPersistence_restoreMessageQueue:918 (0)
20231222 104739.395 (7987781312) (0)< MQTTClient_createWithOptions:522 (0)
20231222 104739.395 (7987781312) (0)> MQTTClient_setCallbacks:1074
20231222 104739.395 (7987781312) (0)< MQTTClient_setCallbacks:1088 (0)
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 315
[10:47:39.395][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 333
20231222 104739.395 (7987781312) (0)> MQTTClient_connectAll:1745
20231222 104739.395 (7987781312)  (1)> UTF8_validateString:160
20231222 104739.395 (7987781312)   (2)> UTF8_validate:134
20231222 104739.395 (7987781312)   (2)< UTF8_validate:146 (1)
20231222 104739.395 (7987781312)  (1)< UTF8_validateString:165 (1)
20231222 104739.395 (7987781312)  (1)> UTF8_validateString:160
20231222 104739.395 (7987781312)   (2)> UTF8_validate:134
20231222 104739.395 (7987781312)   (2)< UTF8_validate:146 (1)
20231222 104739.395 (7987781312)  (1)< UTF8_validateString:165 (1)
20231222 104739.395 (7987781312)  (1)> MQTTClient_connectURI:1504
20231222 104739.395 (7987781312)   (2)> MQTTStrncpy:1085
20231222 104739.395 (7987781312)   (2)< MQTTStrncpy:1095
20231222 104739.395 (7987781312)   (2)> MQTTStrncpy:1085
20231222 104739.395 (7987781312)   (2)< MQTTStrncpy:1095
20231222 104739.395 (7987781312)   (2)> MQTTStrncpy:1085
20231222 104739.395 (7987781312)   (2)< MQTTStrncpy:1095
20231222 104739.395 (7987781312)   (2)> MQTTClient_connectURIVersion:1209
20231222 104739.395 (7987781312)    (3)> Paho_thread_start:68
20231222 104739.395 (7987781312)    (3)< Paho_thread_start:79
20231222 104739.395 (7987781312)    (3)> MQTTTime_sleep:29
20231222 104739.395 (6094286848) (0)> MQTTClient_run:828
20231222 104739.395 (6094286848)  (1)> Thread_set_name:90
20231222 104739.395 (6094286848)  (1)< Thread_set_name:111 (0)
20231222 104739.395 (6094286848)  (1)> MQTTClient_cycle:2566
20231222 104739.395 (6094286848)   (2)> Socket_getReadySocket:495
20231222 104739.395 (6094286848)   (2)< Socket_getReadySocket:604 (0)
20231222 104739.395 (6094286848)   (2)> MQTTClient_retry:2545
20231222 104739.395 (6094286848)    (3)> MQTTProtocol_keepalive:704
20231222 104739.395 (6094286848)    (3)< MQTTProtocol_keepalive:781
20231222 104739.395 (6094286848)    (3)> MQTTProtocol_retry:905
20231222 104739.395 (6094286848)    (3)< MQTTProtocol_retry:924
20231222 104739.395 (6094286848)   (2)< MQTTClient_retry:2555
20231222 104739.395 (6094286848)  (1)< MQTTClient_cycle:2650 (0)
20231222 104739.395 (6094286848)  (1)> MQTTClient_cycle:2566
20231222 104739.395 (6094286848)   (2)> Socket_getReadySocket:495
20231222 104739.395 (6094286848)   (2)< Socket_getReadySocket:604 (0)
20231222 104739.395 (6094286848)   (2)> MQTTTime_sleep:29
20231222 104739.395 (7987781312)    (3)< MQTTTime_sleep:35
20231222 104739.395 Connecting to serverURI localhost with MQTT version 4
20231222 104739.395 (7987781312)    (3)> MQTTProtocol_connect:227
20231222 104739.395 (7987781312)     (4)> MQTTProtocol_addressPort:63
20231222 104739.395 (7987781312)     (4)< MQTTProtocol_addressPort:101
20231222 104739.395 (7987781312)     (4)> Socket_new:1072
20231222 104739.395 New socket 7 for localhost, port 1883
20231222 104739.395 (7987781312)      (5)> Socket_addSocket:272
20231222 104739.395 (7987781312)       (6)> Socket_setnonblocking:95
20231222 104739.395 (7987781312)       (6)< Socket_setnonblocking:100 (0)
20231222 104739.395 (7987781312)      (5)< Socket_addSocket:314 (0)
20231222 104739.395 Connect pending
20231222 104739.395 (7987781312)     (4)< Socket_new:1235 (36)
20231222 104739.395 (7987781312)    (3)< MQTTProtocol_connect:355 (36)
20231222 104739.395 (7987781312)    (3)> MQTTClient_waitfor:2662
20231222 104739.395 (7987781312)     (4)> Thread_wait_sem:273
20231222 104739.497 (7987781312)     (4)< Thread_wait_sem:306 (0)
20231222 104739.497 (7987781312)    (3)< MQTTClient_waitfor:2764 (0)
20231222 104739.497 (7987781312)    (3)> MQTTPacket_send_connect:55
20231222 104739.497 (7987781312)     (4)> MQTTPacket_send:201
20231222 104739.497 (7987781312)      (5)> MQTTPacket_encode:304
20231222 104739.497 (7987781312)      (5)< MQTTPacket_encode:317 (1)
20231222 104739.497 (7987781312)      (5)> MQTTPacket_encode:304
20231222 104739.497 (7987781312)      (5)< MQTTPacket_encode:317 (1)
20231222 104739.497 (7987781312)      (5)> WebSocket_putdatas:944
20231222 104739.497 (7987781312)       (6)> Socket_putdatas:797
20231222 104739.497 (7987781312)        (7)> Socket_writev:721
20231222 104739.497 Socket error Socket is not connected(57) in writev - putdatas for socket 7
20231222 104739.497 (7987781312)        (7)< Socket_writev:773 (-1)
20231222 104739.497 (7987781312)       (6)< Socket_putdatas:852 (-1)
20231222 104739.497 (7987781312)      (5)< WebSocket_putdatas:975 (-1)
20231222 104739.497 (7987781312)     (4)< MQTTPacket_send:236 (-1)
20231222 104739.497 7 Alvaros-MBP.local -> CONNECT version 4 clean: 1 (-1)
20231222 104739.497 (7987781312)    (3)< MQTTPacket_send_connect:129 (-1)
20231222 104739.497 (7987781312)    (3)> MQTTClient_disconnect1:1929
20231222 104739.497 (7987781312)     (4)> MQTTClient_closeSession:1095
20231222 104739.497 (7987781312)      (5)> WebSocket_close:524
20231222 104739.497 (7987781312)      (5)< WebSocket_close:572
20231222 104739.497 (7987781312)      (5)> Socket_close:967
20231222 104739.497 (7987781312)       (6)> Socket_close_only:893
20231222 104739.497 (7987781312)       (6)< Socket_close_only:907 (0)
20231222 104739.497 (7987781312)       (6)> Socket_abortWrite:1355
20231222 104739.497 (7987781312)       (6)< Socket_abortWrite:1376 (0)
20231222 104739.497 (7987781312)       (6)> SocketBuffer_cleanup:153
20231222 104739.497 (7987781312)       (6)< SocketBuffer_cleanup:165
20231222 104739.497 Removed socket 7
20231222 104739.497 Removed socket 7
20231222 104739.497 (7987781312)      (5)< Socket_close:1037 (0)
20231222 104739.497 (7987781312)      (5)> MQTTClient_cleanSession:1131
20231222 104739.497 (7987781312)       (6)> MQTTPersistence_clear:170
20231222 104739.497 (7987781312)       (6)< MQTTPersistence_clear:174 (0)
20231222 104739.497 (7987781312)       (6)> MQTTProtocol_emptyMessageList:1001
20231222 104739.497 (7987781312)       (6)< MQTTProtocol_emptyMessageList:1010
20231222 104739.497 (7987781312)       (6)> MQTTProtocol_emptyMessageList:1001
20231222 104739.497 (7987781312)       (6)< MQTTProtocol_emptyMessageList:1010
20231222 104739.497 (7987781312)       (6)> MQTTClient_emptyMessageQueue:557
20231222 104739.497 (7987781312)       (6)< MQTTClient_emptyMessageQueue:572
20231222 104739.497 (7987781312)      (5)< MQTTClient_cleanSession:1139 (0)
20231222 104739.497 (7987781312)     (4)< MQTTClient_closeSession:1123
20231222 104739.497 (7987781312)    (3)< MQTTClient_disconnect1:1963 (0)
20231222 104739.497 (7987781312)   (2)< MQTTClient_connectURIVersion:1477 (-1)
20231222 104739.497 (7987781312)   (2)> MQTTClient_connectURIVersion:1209
20231222 104739.497 Connecting to serverURI localhost with MQTT version 3
20231222 104739.497 (7987781312)    (3)> MQTTProtocol_connect:227
20231222 104739.497 (7987781312)     (4)> MQTTProtocol_addressPort:63
20231222 104739.497 (7987781312)     (4)< MQTTProtocol_addressPort:101
20231222 104739.497 (7987781312)     (4)> Socket_new:1072
20231222 104739.497 New socket 7 for localhost, port 1883
20231222 104739.497 (7987781312)      (5)> Socket_addSocket:272
20231222 104739.497 (7987781312)       (6)> Socket_setnonblocking:95
20231222 104739.497 (7987781312)       (6)< Socket_setnonblocking:100 (0)
20231222 104739.497 (7987781312)      (5)< Socket_addSocket:314 (0)
20231222 104739.497 Connect pending
20231222 104739.497 (7987781312)     (4)< Socket_new:1235 (36)
20231222 104739.497 (7987781312)    (3)< MQTTProtocol_connect:355 (36)
20231222 104739.497 (7987781312)    (3)> MQTTClient_waitfor:2662
20231222 104739.498 (7987781312)     (4)> Thread_wait_sem:273
20231222 104739.498 (7987781312)     (4)< Thread_wait_sem:306 (0)
20231222 104739.498 (7987781312)    (3)< MQTTClient_waitfor:2764 (0)
20231222 104739.498 (7987781312)    (3)> MQTTPacket_send_connect:55
20231222 104739.498 (7987781312)     (4)> MQTTPacket_send:201
20231222 104739.498 (7987781312)      (5)> MQTTPacket_encode:304
20231222 104739.498 (7987781312)      (5)< MQTTPacket_encode:317 (1)
20231222 104739.498 (7987781312)      (5)> MQTTPacket_encode:304
20231222 104739.498 (7987781312)      (5)< MQTTPacket_encode:317 (1)
20231222 104739.498 (7987781312)      (5)> WebSocket_putdatas:944
20231222 104739.498 (7987781312)       (6)> Socket_putdatas:797
20231222 104739.498 (7987781312)        (7)> Socket_writev:721
20231222 104739.498 Socket error Socket is not connected(57) in writev - putdatas for socket 7
20231222 104739.498 (7987781312)        (7)< Socket_writev:773 (-1)
20231222 104739.498 (7987781312)       (6)< Socket_putdatas:852 (-1)
20231222 104739.498 (7987781312)      (5)< WebSocket_putdatas:975 (-1)
20231222 104739.498 (7987781312)     (4)< MQTTPacket_send:236 (-1)
20231222 104739.498 7 Alvaros-MBP.local -> CONNECT version 3 clean: 1 (-1)
20231222 104739.498 (7987781312)    (3)< MQTTPacket_send_connect:129 (-1)
20231222 104739.498 (7987781312)    (3)> MQTTClient_disconnect1:1929
20231222 104739.498 (7987781312)     (4)> MQTTClient_closeSession:1095
20231222 104739.498 (7987781312)      (5)> WebSocket_close:524
20231222 104739.498 (7987781312)      (5)< WebSocket_close:572
20231222 104739.498 (7987781312)      (5)> Socket_close:967
20231222 104739.498 (7987781312)       (6)> Socket_close_only:893
20231222 104739.498 (7987781312)       (6)< Socket_close_only:907 (0)
20231222 104739.498 (7987781312)       (6)> Socket_abortWrite:1355
20231222 104739.498 (7987781312)       (6)< Socket_abortWrite:1376 (0)
20231222 104739.498 (7987781312)       (6)> SocketBuffer_cleanup:153
20231222 104739.498 (7987781312)       (6)< SocketBuffer_cleanup:165
20231222 104739.498 Removed socket 7
20231222 104739.498 Removed socket 7
20231222 104739.498 (7987781312)      (5)< Socket_close:1037 (0)
20231222 104739.498 (7987781312)      (5)> MQTTClient_cleanSession:1131
20231222 104739.498 (7987781312)       (6)> MQTTPersistence_clear:170
20231222 104739.498 (7987781312)       (6)< MQTTPersistence_clear:174 (0)
20231222 104739.498 (7987781312)       (6)> MQTTProtocol_emptyMessageList:1001
20231222 104739.498 (7987781312)       (6)< MQTTProtocol_emptyMessageList:1010
20231222 104739.498 (7987781312)       (6)> MQTTProtocol_emptyMessageList:1001
20231222 104739.498 (7987781312)       (6)< MQTTProtocol_emptyMessageList:1010
20231222 104739.498 (7987781312)       (6)> MQTTClient_emptyMessageQueue:557
20231222 104739.498 (7987781312)       (6)< MQTTClient_emptyMessageQueue:572
20231222 104739.498 (7987781312)      (5)< MQTTClient_cleanSession:1139 (0)
20231222 104739.498 (7987781312)     (4)< MQTTClient_closeSession:1123
20231222 104739.498 (7987781312)     (4)> MQTTClient_stop:1028
20231222 104739.498 Conn_count is 0
20231222 104739.498 sleeping
20231222 104739.498 (7987781312)      (5)> MQTTTime_sleep:29
20231222 104739.498 (6094286848)   (2)< MQTTTime_sleep:35
20231222 104739.498 (6094286848)   (2)> MQTTClient_retry:2545
20231222 104739.498 (6094286848)    (3)> MQTTProtocol_retry:905
20231222 104739.498 (6094286848)    (3)< MQTTProtocol_retry:924
20231222 104739.498 (6094286848)   (2)< MQTTClient_retry:2555
20231222 104739.498 (6094286848)  (1)< MQTTClient_cycle:2650 (0)
20231222 104739.498 (6094286848) (0)< MQTTClient_run:1016
20231222 104739.498 (7987781312)      (5)< MQTTTime_sleep:35
20231222 104739.498 (7987781312)     (4)< MQTTClient_stop:1063 (1)
20231222 104739.498 (7987781312)    (3)< MQTTClient_disconnect1:1963 (0)
20231222 104739.498 (7987781312)   (2)< MQTTClient_connectURIVersion:1477 (-1)
20231222 104739.498 (7987781312)  (1)< MQTTClient_connectURI:1702 (-1)
20231222 104739.599 (7987781312) (0)< MQTTClient_connectAll:1910 (-1)
[10:47:39.599][252504/252504  0] MQTT: Failed to connect, return code -1
[10:47:39.599][252504/252504  0] MQTT: src/mqtt/hasp_mqtt_paho_single.cpp 346
[10:47:39.599][252504/252504  0] MAIN: src/main_sdl2.cpp 137
[10:47:39.599][252504/252504  0] GPIO: Saving Pin config #0 pin 3 - type 11 - group 0 - func 255
[10:47:39.599][252504/252504  0] GPIO: Saving Pin config #1 pin 4 - type 10 - group 0 - func 255
[10:47:39.599][252504/252504  0] GPIO: Saving Pin config #2 pin 13 - type 2 - group 0 - func 255
[10:47:39.599][252504/252504  0] GPIO: Saving Pin config #3 pin 14 - type 34 - group 0 - func 255
[10:47:39.599][252504/252504  0] GPIO: Saving Pin config #4 pin 5 - type 172 - group 0 - func 255
[10:47:39.599][252504/252504  0] MAIN: src/main_sdl2.cpp 146
[10:47:39.599][252504/252504  0] MAIN: loop started
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
20231222 104739.599 (7987781312) (0)> MQTTClient_receive:2777
20231222 104739.599 (7987781312) (0)< MQTTClient_receive:2820 (-3)
```

**Comment by alvarolobato** [comment]
Taking a look at the changes between the two versions it looks like this change is what broke it:
https://github.com/HASwitchPlate/openHASP/commit/56ae45dae251566e38fc72e42395282d62e12a13#diff-979b76318c39babb49890eba446d0fc0e22c2cace106834c8eb54dd873c026b8L307

Commenting back again the method to add the callback made it work again.

**Comment by fvanroie** [comment]
Can you create a PR with the fix, please?

**Comment by fvanroie** [comment]
I believe this has been fixed inPR #631 and #640, can you confirm this is working for you now?
